### PR TITLE
feat: add support for DID Metadata

### PR DESCRIPTION
**Pull request by noandrea**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -22,10 +22,11 @@ require (
 	github.com/tendermint/tendermint v0.34.8
 	github.com/tendermint/tm-db v0.6.4
 	github.com/wealdtech/go-merkletree v1.0.0
-	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83 // indirect
+	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
 	golang.org/x/term v0.0.0-20201210144234-2321bbc49cbf // indirect
 	google.golang.org/genproto v0.0.0-20210223151946-22b48be4551b
 	google.golang.org/grpc v1.36.0
+	google.golang.org/protobuf v1.25.0
```
Comment (alessio) [inline go.mod:29]: I'd bump the SDK dependency to the v0.42.LATEST and ensure the protobuf dependency matches what's used by the SDK's latest release (it should be v1.26.0)

**Review by alessio — COMMENTED** [review]
I left a small comment. Looks good to me otherwise

File: go.mod (hunk 1 above)
Comment (noandrea) [reply go.mod:29]: I'm going to open a pr dedicated to bumping version. there are some breaking changes in the 0.42.9 and I had trouble dealing with the intermediate versions on different branches.
Once we have the features on main (mainly this and the #153 swagger one) I'll got for the 0.43.0

File: x/did/keeper/did.go [hunk 2]
```diff
@@ -1,63 +1,90 @@
...
+func (k Keeper) UnmarshalDidMetadata(value []byte) (interface{}, bool) {
+	data := types.DidMetadata{}
+	k.Unmarshal(value, &data)
+	return data, types.IsValidDIDMetadata(&data)
+}
 
-	bytes, _ := k.cdc.MarshalBinaryBare(&did)
+func (k Keeper) Marshal(value interface{}) (bytes []byte) {
+	switch value := value.(type) {
+	case types.DidDocument:
```
Comment (jgimeno) [inline x/did/keeper/did.go:44]: If I am not mistaken you can use something on the style of case types.DidDocument, types.DidMetadata: and you can avoid repeating the encoding line.

Comment (noandrea) [reply x/did/keeper/did.go:44]: that is automatically casting `value` to the type in the  `case`,  with multiple case options the automatic casting will not work and value will be just an `interface{}`

**Review by alessio — APPROVED** [review]

**Pull request merged**